### PR TITLE
ui/issue 31

### PR DESCRIPTION
**Pull request by iSkore**
Showing previous user annotations per image
Showing different colors per user

### CODE REVIEW

File: apps/tlv-app/grails-app/assets/javascripts/menus/annotations-map.js [hunk 1]
```diff
@@ -397,146 +421,318 @@ function saveAnnotations() {
...
+		var style      = properties.style || createDefaultStyle();
+		var label      = properties.type + '\n' + properties.username;
+		
+		style.getText().setText( label );
+		feature.setStyle( style );
+	} );
+}
 
+function generateRandomNumber( min, max ) {
+	return ~~( Math.random() * max ) + min;
```
Comment (cdowin) [inline apps/tlv-app/grails-app/assets/javascripts/menus/annotations-map.js]:
```suggestion
	return Math.floor(Math.random() * (max - min) ) + min;
```

Comment (cdowin) [reply apps/tlv-app/grails-app/assets/javascripts/menus/annotations-map.js]: Remove floor() if you don't want an integer. The max - min math avoids the possibility of generating a number greater than the MAX if min=max

File: apps/tlv-app/grails-app/assets/javascripts/menus/annotations-map.js [hunk 2]
```diff
@@ -552,17 +748,19 @@ function bindWindowUnload() {
...
 }
 
 var setupTimeLapseAnnotations = setupTimeLapse;
-setupTimeLapse = function() {
+setupTimeLapse                = function() {
 	setupTimeLapseAnnotations();
-	//searchForAnnotations();
+	
+	searchForAnnotations();
+	// annotationsLayerToggle();
```
Comment (cdowin) [inline apps/tlv-app/grails-app/assets/javascripts/menus/annotations-map.js]:
```suggestion
```

Comment (cdowin) [reply apps/tlv-app/grails-app/assets/javascripts/menus/annotations-map.js]: Remove dead code

File: apps/tlv-app/grails-app/assets/javascripts/time-lapse-map.js [hunk 3]
```diff
@@ -165,7 +165,8 @@ function createMapControls() {
 		var button = document.createElement( "button" );
 		button.innerHTML = "<span class = 'glyphicon glyphicon-step-forward'></span>";
 		button.title = "Fast Forward";
-
+		button.id = "fastForward";
```
Comment (knaquin) [inline apps/tlv-app/grails-app/assets/javascripts/time-lapse-map.js:168]: Does this need to have an ID? Is it used anywhere else?

File: apps/tlv-app/grails-app/assets/javascripts/menus/annotations-map.js [hunk 4]
```diff
@@ -95,257 +103,274 @@ function applyAnnotationStyle() {
...
-	$( "#usernameInput" ).val( properties.username );
+	$( '#confidenceSelect option[value=' + properties.confidence + ']' ).prop( 'selected', true );
+	$( '#typeInput' ).val( properties.type );
+	$( '#usernameInput' ).val( properties.username );
 }
 
-var pageLoadAnnotation = pageLoad
-pageLoad = function() {
+var pageLoadAnnotation = pageLoad;
+pageLoad               = function() {
```
Comment (knaquin) [inline apps/tlv-app/grails-app/assets/javascripts/menus/annotations-map.js:359]: Too many spaces? Probably something I did.

File: apps/tlv-app/grails-app/assets/javascripts/menus/annotations-map.js [hunk 5]
```diff
@@ -499,78 +524,246 @@ function saveAnnotations() {
...
+						id: 'annotation.' + annotation.id,
+						type: 'Feature',
+						properties: annotation
+					};
+					
+					if( !usernames.hasOwnProperty( annotation.username ) ) {
+						usernames[ annotation.username ] = generateRandomStyle( 140, 255 );
+					}
+					
+					geoJSON.properties.style = usernames[ annotation.username ];
```
Comment (knaquin) [inline apps/tlv-app/grails-app/assets/javascripts/menus/annotations-map.js:713]: Could we only generate random styles if there are annotations from more than 1 user?

File: apps/tlv-app/grails-app/assets/javascripts/menus/annotations-map.js [hunk 6]
```diff
@@ -499,78 +524,246 @@ function saveAnnotations() {
...
+}
+
+function searchForAnnotations() {
+	var usernames = {};
+	
+	tlv.layers.forEach( function( layer ) {
+		console.log( layer );
+		
+		$.ajax( {
+			url: '/annotation/imageAnnotations' + '?' + $.param( { imgId: layer.imageId } )
```
Comment (knaquin) [inline apps/tlv-app/grails-app/assets/javascripts/menus/annotations-map.js:683]: It looks like you eventually create a GeoJSON object from the data anyway, why is a custom endpoint needed to get the data when WFS will do all the conversion work for you?

**Review by knaquin — CHANGES_REQUESTED** [review]
See comments...

File: apps/tlv-app/grails-app/assets/javascripts/menus/annotations-map.js (hunk 4 above)
Comment (iSkore) [reply apps/tlv-app/grails-app/assets/javascripts/menus/annotations-map.js:359]: Nah, that's my editor. Keeps all the `=` inline

File: apps/tlv-app/grails-app/assets/javascripts/menus/annotations-map.js (hunk 6 above)
Comment (iSkore) [reply apps/tlv-app/grails-app/assets/javascripts/menus/annotations-map.js:683]: I was trying to get the annotations applied to an image and display them. Are you saying I can get that from WFS?

Comment (knaquin) [reply apps/tlv-app/grails-app/assets/javascripts/menus/annotations-map.js:683]: Yeah, the WFS will return all the annotations for a specific image in GeoJSON format. The issue, for testing at least, is that the features will be the ones on the server, not the ones you've created locally. But you can always just used the dev instance, create some features, save them and then do all your testing locally.

**Comment by knaquin** [comment]
Request has gone stale.

**Pull request closed without merging**